search-resources-api/src/main/java/no/unit/nva/search/SearchAllHandler.java

```diff
@@ -71,10 +74,20 @@ private static IdentityClient defaultIdentityClient() {
         return new IdentityClientImpl();
     }
 
-    private Integer extractPageSize(RequestInfo requestInfo) {
-        return requestInfo.getQueryParameterOpt(PAGE_SIZE_QUERY_PARAM)
+    private Integer extractResultsSize(RequestInfo requestInfo) {
+        return requestInfo.getQueryParameterOpt(RESULTS_QUERY_PARAM)
             .map(Integer::valueOf)
-            .orElse(DEFAULT_PAGE_SIZE);
+            .orElse(DEFAULT_RESULTS_SIZE);
+    }
+
+    private Integer extractPageNo(RequestInfo requestInfo) {
+        return requestInfo.getQueryParameterOpt(PAGE_QUERY_PARAM)
+            .map(Integer::valueOf)
+            .orElse(DEFAULT_RESULTS_INDEX);
+    }
+
+    private Integer calculateResultsFromIndex(RequestInfo requestInfo) {
+        return extractPageNo(requestInfo) * extractResultsSize(requestInfo);
```

Comment:
Each class should have only one responsibility (Single-responsibility principle). The responsibility of this class is to get the REST request and forward the input parameters to the  associated services. The calculations should happen in the SearchClient and not here. Then you can reuse the code for the SearchHandler that is used for searching publications. SearchAll handler searches DoiRequests and SupportMessages. The class names are a bit wrong